src/main/java/com/societegenerale/commons/plugin/gradle/WorkerActionParams.java

```diff
@@ -1,8 +1,34 @@
 package com.societegenerale.commons.plugin.gradle;
 
+import org.gradle.api.provider.ListProperty;
 import org.gradle.api.provider.Property;
 import org.gradle.workers.WorkParameters;
 
+import java.io.Serializable;
+import java.util.ArrayList;
+
 public interface WorkerActionParams extends WorkParameters {
-    Property<ArchUnitGradleConfig> getArchUnitGradleConfig();
+    Property<String> getMainClassesPath();
+
+    Property<String> getTestClassesPath();
+
+    ListProperty<String> getExcludedPaths();
+
+    ListProperty<String> getPreConfiguredRules();
+
+    ListProperty<ConfigurableRule> getConfigurableRules();
+
+    class ConfigurableRule implements Serializable {
```

Comment:
If `com.societegenerale.commons.plugin.model.ConfigurableRule` was made serializable then these clones could be removed along with all the mapping code I added to transform between them. I don't think that would add too much gradle specific overhead to the common plugin code.

Reply:
If these transport objects are going to be kept, then it would probably be better to refactor them. I just slapped together  something quick using the fields directly :)

Reply:
thanks a lot for the help, I think it really helped me getting to something that seems to work ! 

I can make the classes Serializable to make things simpler (https://github.com/societe-generale/arch-unit-build-plugin-core/compare/master...makingSomeClassesSerializable ) , but I first need to make another change to upgrade the core library in the plugin